x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/api.test.ts

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaServices } from '../../../lib/kibana';
+import {
+  signalsMock,
+  mockSignalsQuery,
+  mockStatusSignalQuery,
+  mockSignalIndex,
+  mockUserPrivilege,
+} from './mock';
+import {
+  fetchQuerySignals,
+  updateSignalStatus,
+  getSignalIndex,
+  getUserPrivilege,
+  createSignalIndex,
+} from './api';
+import { ToasterErrors } from '../../../components/ml/api/throw_if_not_ok';
+import { PostSignalError, SignalIndexError } from './types';
+
+const abortCtrl = new AbortController();
+const mockKibanaServices = KibanaServices.get as jest.Mock;
+jest.mock('../../../lib/kibana');
+
+const mockfetchSuccess = (body: unknown, fetchMock?: jest.Mock) => {
```

Comment:
I cleaned all this mocking up on https://github.com/elastic/kibana/commit/625aa5f2c24d3b306f2213ce6af5ba99d0f73e86; please take a look and feel free to cherry-pick that here.